Assets/MixedRealityToolkit.Tests/PlayModeTests/BoundingBoxTests.cs

```diff
@@ -0,0 +1,282 @@
+
+        /// <summary>
+        /// Uses near interaction to scale the bounding box by directly grabbing corner
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator ScaleViaNearInteration()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            var bounds = bbox.GetComponent<BoxCollider>().bounds;
+            Debug.Assert(bounds.center == new Vector3(0, 0, 1.5f));
+            Debug.Assert(bounds.size == new Vector3(.5f, .5f, .5f));
+
+            List<GameObject> corners = FindDescendantsContainingName(bbox.gameObject, "corner_");
+            var inputSimulationService = PlayModeTestUtilities.GetInputSimulationService();
+            // front right corner is corner 3
+            var frontRightCornerPos = corners[3].transform.position;
+
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSimulationService, ArticulatedHandPose.GestureId.Open, frontRightCornerPos);
+            yield return PlayModeTestUtilities.SetHandState(frontRightCornerPos, ArticulatedHandPose.GestureId.Pinch, Handedness.Right, inputSimulationService);
+            var delta = new Vector3(0.1f, 0.1f, 0f);
+            yield return PlayModeTestUtilities.MoveHandFromTo(frontRightCornerPos, frontRightCornerPos + delta, 10, ArticulatedHandPose.GestureId.Pinch, Handedness.Right, inputSimulationService);
+            var endBounds = bbox.GetComponent<BoxCollider>().bounds;
+            TestUtilities.AssertAboutEqual(endBounds.center, new Vector3(0.033f, 0.033f, 1.467f), "endBounds incorrect center");
+            TestUtilities.AssertAboutEqual(endBounds.size, Vector3.one * .567f, "endBounds incorrect size");
+
+            GameObject.Destroy(bbox.gameObject);
+            // Wait for a frame to give Unity a change to actually destroy the object
+            yield return null;
+
+        }
+
+        /// <summary>
+        /// Uses far interaction (HoloLens 1 style) to scale the bounding box
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator ScaleViaHoloLens1Interaction()
+        {
+            var bbox = InstantiateSceneAndDefaultBbox();
+            yield return null;
+            var bounds = bbox.GetComponent<BoxCollider>().bounds;
+            var startCenter = new Vector3(0, 0, 1.5f);
+            var startSize = new Vector3(.5f, .5f, .5f);
+            TestUtilities.AssertAboutEqual(bounds.center, startCenter, "bbox incorrect center at start");
+            TestUtilities.AssertAboutEqual(bounds.size, startSize, "bbox incorrect size at start");
+
+            var iss = PlayModeTestUtilities.GetInputSimulationService();
+            var oldIsp = iss.InputSimulationProfile;
+            var isp = new MixedRealityInputSimulationProfile
+            {
+                HandSimulationMode = HandSimulationMode.Gestures
+            };
+            iss.InputSimulationProfile = isp;
```

Comment:
So swapping out just the input sim profile works like this? I tried something similar for writing gesture event tests, but was setting the entire MRTK.Instance.ActiveProfile because i didn't want to risk changing assets (the default input system profile) just by running tests. If the tests fail they would not reset the profile and inadvertently change the asset, which might get saved automatically by Unity. But changing the ActiveProfile at runtime causes a lot of broken behavior. Just changing the input sim profile probably is easier to keep working.

Reply:
Yep, this was working for me! The only thing I needed to make sure was to change the profile back to original at the end of the test.